### DISCUSSION TITLE
PATH on Windows - Conda/Mamba environment issues

### DISCUSSION THREAD

**Discussion by drewbitt** [evilmartians/lefthook#438] [General]
Hi,

Conda/Mamba, the package/dependency/environment manager for Python, creates Python environments. This allows for switching python versions, paths, and packages via `conda activate my-env`. This is typically done in every shell as the first command. Conda does this via path manipulation hooks, e.g. in Powershell this is 

`(& "C:\Users\user\condapath\conda.exe" "shell.powershell" "hook") | Out-String | ?{$_} | Invoke-Expression` 

Once you have activated your environment, the resulting path additions in Windows looks like
```
C:\Users\andys-pc\scoop\apps\mambaforge\current\envs\my-env;C:\Users\andys-pc\scoop\apps\mambaforge\current\envs\my-env\Library\mingw-w64\bin;C:\Users\andys-pc\scoop\apps\mambaforge\current\envs\my-env\Library\usr\bin;C:\Users\andys-pc\scoop\apps\mambaforge\current\envs\my-env\Library\bin;C:\Users\andys-pc\scoop\apps\mambaforge\current\envs\my-env\Scripts;C:\Users\andys-pc\scoop\apps\mambaforge\current\envs\my-env\bin;C:\Users\andys-pc\scoop\apps\mambaforge\current\condabin
```

binaries can be installed via `conda`, `mamba`, or `pip`, which will be included in the PATH of your shell post-activation. But when executing lefthook as a pre-commit hook, it will no longer be in your PATH, nor can I figure out how to activate conda successfully in order to get this PATH injection for lefthook to find binaries I would like to run. 

Is there any example of getting conda/mamba paths injected into lefthook before running pre-commit hooks?

**Comment by mrexox** [comment]
Hey! Yes, there is a [`rc`](https://github.com/evilmartians/lefthook/blob/master/docs/configuration.md#rc) configuration option which allows you to use a sh syntax to extend the environment. There's also an [`env`](https://github.com/evilmartians/lefthook/blob/master/docs/configuration.md#env) option for specific commands/scripts but it is not so flexible (although can be used to [extend PATH variable](https://github.com/evilmartians/lefthook/blob/master/docs/configuration.md#extending-path))

I haven't tested `rc` option on Windows :confused: Please write back if it worked for you :pray:
